coresimd/src/x86/mod.rs

```diff
@@ -8,6 +10,70 @@ mod macros;
 #[allow(non_camel_case_types)]
 pub struct __m128(f32, f32, f32, f32);
 
+#[repr(simd)]
+#[derive(Clone, Copy, Debug)]
+#[allow(non_camel_case_types)]
+pub struct __m128d(f64, f64);
+
+pub use v128::__m128i;
+pub use v64::__m64;
+
+#[cfg(test)]
+mod test;
+#[cfg(test)]
+pub use self::test::*;
+
+#[doc(hidden)]
+#[allow(non_camel_case_types)]
+trait m128iExt: Sized {
```

Comment:
This is somewhat smaller than what the macros were doing, but we'll how it evolves as the avx intrinsics get ported.

Reply:
Yeah this was just what I felt was needed to define the various intrinsics, although it may be a better idea to separate out what these are converting too from the actual `::simd` portable types (they don't need to be the same)